cypress-tests/cypress/e2e/v2/projectResources.cy.ts

```diff
@@ -140,6 +136,173 @@ describe("Project resources - work with code, data, environments", () => {
+      .contains(name)
+      .click();
+
+    // Open the dropdown menu and click delete
+    cy.getDataCy("data-connector-menu-dropdown")
+      .should("be.visible")
+      .scrollIntoView()
+      .click();
+    cy.getDataCy("data-connector-delete").should("be.visible").click();
+
+    // Confirm deletion by typing the slug
+    cy.getDataCy("delete-confirmation-input").type(name);
+    cy.getDataCy("delete-data-connector-modal-button").click();
+
+    // Verify the data connector is deleted
+    cy.getDataCy("data-connector-box")
+      .find(`[data-cy=data-connector-name]`)
+      .contains(name)
+      .should("not.exist");
+  });
+
+  it("Link an existing data connector to a project", () => {
+    const name = `giab-link-${getRandomString()}`;
+
+    // Create a data connector not linked to a project
+    getUserData().then((user: User) => {
+      const slug = name;
+      const dataConnectorIdentifier = `${user.username}/${slug}`;
+
+      // Navigate to the user's namespace page by clicking the namespace link
+      visitCurrentProject();
+      cy.getDataCy("project-namespace-link").click();
+
+      cy.getDataCy("add-data-connector").click();
+      cy.getDataCy("data-storage-s3").click();
+      cy.getDataCy("data-provider-AWS").click();
+      cy.getDataCy("data-connector-edit-next-button").click();
+
+      cy.getDataCy("data-connector-source-path")
+        .should("be.empty")
+        .type("giab");
+      cy.getDataCy("test-data-connector-button").click();
+      cy.getDataCy("cloud-storage-connection-success").should("be.visible");
+      cy.getDataCy("add-data-connector-continue-button").click();
+
+      cy.getDataCy("data-connector-name-input").should("be.empty").type(name);
+      cy.getDataCy("data-connector-slug-toggle").click();
+      cy.getDataCy("data-connector-slug-input").should("have.value", name);
+      cy.getDataCy("data-connector-edit-update-button").click();
+      cy.getDataCy("data-connector-edit-success").should("be.visible");
+      cy.getDataCy("data-connector-edit-close-button").click();
+
+      // Now link the data connector to the project
+      visitCurrentProject();
+      cy.getDataCy("add-data-connector").click();
+      cy.getDataCy("project-data-controller-mode-link").click();
+
+      // Enter the data connector identifier
+      cy.get("#data-connector-identifier")
+        .should("be.empty")
+        .type(dataConnectorIdentifier);
+      cy.getDataCy("link-data-connector-button").click();
+
+      // ? Currently, data connectors newly linked might not appear immediately
+      visitCurrentProject();
+      cy.getDataCy("data-connector-box")
+        .find(`[data-cy=data-connector-name]`)
+        .contains(name);
+    });
+  });
+
+  it("Unlink a data connector from a project", () => {
+    const name = `giab-link-${getRandomString()}`;
+
+    // Create a data connector not linked to a project
+    getUserData().then((user: User) => {
+      const slug = name;
```

Comment:
Nitpick: nothing wrong here, but you use `slug` only on the next line, and then use `name` in all the following occurrences. You could drop this line and use `name` as well in the next line